resources/api.go

```diff
@@ -0,0 +1,120 @@
+package resources
+
+import "github.com/nitrictech/go-sdk/faas"
+
+type Route interface {
+	Get(handler ...faas.HttpMiddleware) error
+	Patch(handler ...faas.HttpMiddleware) error
+	Put(handler ...faas.HttpMiddleware) error
+	Post(handler ...faas.HttpMiddleware) error
+	Delete(handler ...faas.HttpMiddleware) error
+}
+
+type route struct {
+	opts           faas.ApiWorkerOptions
+	handlerBuilder faas.HandlerBuilder
+}
+
+func NewRoute(apiName, path string) Route {
+	return &route{
+		opts: faas.ApiWorkerOptions{
+			ApiName:     apiName,
+			Path:        path,
+			HttpMethods: []string{},
+		},
+		handlerBuilder: faas.New(),
+	}
+}
+
+func (r *route) optsWithMethod(method string) faas.ApiWorkerOptions {
+	r.opts.HttpMethods = append(r.opts.HttpMethods, method)
+	return r.opts
+}
+
+func (r *route) Get(handlers ...faas.HttpMiddleware) error {
+	return r.handlerBuilder.Http(handlers...).WithApiWorkerOpts(r.optsWithMethod("GET")).Start()
+}
+
+func (r *route) Post(handlers ...faas.HttpMiddleware) error {
+	return r.handlerBuilder.Http(handlers...).WithApiWorkerOpts(r.optsWithMethod("POST")).Start()
+}
+
+func (r *route) Put(handlers ...faas.HttpMiddleware) error {
+	return r.handlerBuilder.Http(handlers...).WithApiWorkerOpts(r.optsWithMethod("PUT")).Start()
+}
+
+func (r *route) Patch(handlers ...faas.HttpMiddleware) error {
+	return r.handlerBuilder.Http(handlers...).WithApiWorkerOpts(r.optsWithMethod("PATCH")).Start()
+}
+
+func (r *route) Delete(handlers ...faas.HttpMiddleware) error {
+	return r.handlerBuilder.Http(handlers...).WithApiWorkerOpts(r.optsWithMethod("DELETE")).Start()
+}
```

Comment:
This is the part where we'll run into issues with worker registration and blocking:

```go
func main() {
  mainApi := nitric.Api("main")
 
  // registers at deploy time
  // blocks at runtime (still blocks at deploytime but the deployment server immediately terminates the stream)
  mainApi.Get("/route1", func(...) { ... })

  // registers at deploy time
  // will never be accessible at runtime (unless the above worker errors).
  mainApi.Get("/route2", func(...) { ... })
}
```

Ultimately at runtime we still need to block on something and the developer will need to include this somewhere. 
@asalkeld it would be good to get your thoughts on how this should look.

We could potentially delay the `start` of workers and group them together with a waitgroup (exit when `n` number of workers fail).

```go
func main() {
  mainApi := nitric.Api("main")
 
  // registers at deploy time
  // blocks at runtime (still blocks at deploytime but the deployment server immediately terminates the stream)
  mainApi.Get("/route1", func(...) { ... })

  // registers at deploy time
  // will never be accessible at runtime (unless the above worker errors).
  mainApi.Get("/route2", func(...) { ... })
  
  // Block and start workers
  nitric.Start()
}
```

We could also start the workers in goroutines as soon as they are called and collect error channels from their and block using those channels.

Let me know what you think.

Reply:
Just pushed a version that hopefully does the above now..